instrumentation/instapgx/conn.go

```diff
@@ -0,0 +1,216 @@
+// (c) Copyright IBM Corp. 2021
+// (c) Copyright Instana Inc. 2020
+
+//go:build go1.13
+// +build go1.13
+
+package instapgx
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	instana "github.com/instana/go-sensor"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4"
+)
+
+var detailedBatchMode = false
+var detailedBatchModeMaxEntries = 10
+
+// EnableDetailedBatchMode allows reporting up to detailedBatchModeMaxEntries sql when executing sql in batches.
+func EnableDetailedBatchMode() {
+	detailedBatchMode = true
+}
+
+// Conn wraps *pgx.Conn and adds tracing to it.
+type Conn struct {
+	*pgx.Conn
+	sensor *instana.Sensor
+	config *pgx.ConnConfig
+}
+
+// Prepare wraps (*pgx.Conn).Prepare method and adds tracing to it.
+func (c *Conn) Prepare(ctx context.Context, name, sql string) (*pgconn.StatementDescription, error) {
+	childCtx, span := contextWithChildSpan(ctx, fmt.Sprintf("PREPARE %s AS %s", name, sql), c.config, c.sensor)
```

Comment:
Do we need to worry about sensitive data that needs to be redacted from `sql`?
For instance, if the contents of the `sql` variable are displayed in the UI or somewhere else, where values from a query could be visible.

Reply:
We do not collect parameters for a statement. In case it is embedded into SQL string then it will be there.